src/core/border_router/routing_manager.cpp

```diff
@@ -1771,6 +1781,11 @@ TimeMilli RoutingManager::DiscoveredPrefixTable::Entry::GetStaleTime(void) const
     return mLastUpdateTime + TimeMilli::SecToMsec(delay);
 }
 
+TimeMilli RoutingManager::DiscoveredPrefixTable::Entry::GetStaleTimeFromPrefferedLifetime(void) const
+{
+    return mLastUpdateTime + CalculateExpireDelay(GetPreferredLifetime());
+}
```

Comment:
Can we reuse the `GetStaleTime()`? 
I think `PrefixPd` is always marked as `OnLinkPrefix` so would use the `GetPreferredLifetime()`?

Reply:
Thanks. I see the `kRtrAdvStaleTime` wont be applicable in this case.

So let's keep the two methods. 

👍  on using `CalculateExpireDelay()`.